lib/core/src/connection_pool.cpp

```diff
@@ -237,11 +281,39 @@ namespace irods
         }
 
         try {
-            // NOLINTNEXTLINE(bugprone-unused-raii)
-            query<RcComm>{ctx.conn.get(), "select ZONE_NAME where ZONE_TYPE = 'local'"};
+            // Check if the number of requests executed via this connection has been reached.
```

Comment:
Please update this comment as well to reflect the behavior, if necessary.

Reply:
Done